src/status_im/ui/screens/profile/visibility_status/utils.cljs

```diff
@@ -100,12 +100,12 @@
 
 (defn icon-visibility-status-dot
   [public-key container-size]
-  (let [my-icon?                 (my-icon? public-key)
-        visibility-status-update (visibility-status-update public-key my-icon?)
-        size                     (icon-dot-size container-size)
-        margin                   -2
-        dot-color                (icon-dot-color visibility-status-update)
-        new-ui?                  true]
+  (let [icon?     (my-icon? public-key)
```

Comment:
Hi @ilmotta, Thank you very much for your awesome work :rocket: 

Not sure what will be the perfect name here, but here `my-icon?` function checks if the visibility status update is being shown for the current user avatar. So `icon?` is a little confusing, maybe we rename the original function to something like my-public-key? or something else. wdyt?

Reply:
Please also rename in function `visibility-status-update` or maybe we can get rid of this function completely and move this check there.

Reply:
@Parveshdhull, makes sense. So I deleted the function to check the public key and moved it's code to `visibility-status-update` https://github.com/status-im/status-mobile/pull/17138/commits/a7fad5a74b5f1152b30f4072b0dcedd9b61140ae. I prefer to avoid making further changes to stay within the PR's scope, but I think `visibility-status-update` should be a subscription. Every time I see a function in the view layer treating subs as if they're normal functions it's a probable indicator that it should be extracted into the subscription layer (if there are no other constraints, like performance for some reason).

Reply:
Looks like we already have this check in subscription, (except for empty public-key)
https://github.com/status-im/status-mobile/blob/141bf5a333c0c2e6d66da4a23d2b6b2f6de1ec03/src/status_im2/subs/general.cljs#L16

Reply:
Above check is probably outdated

Reply:
That's good! It seems safe to replace the function `visibility-status-update` with that subscription then.

Reply:
feel free to get rid of `visibility-status-update` function or we can do that later.

Reply:
I can add the check for emptiness, seems like a harmless check. Wdyt?

Reply:
Done. Works well in my tests with 2 users. https://github.com/status-im/status-mobile/pull/17138/commits/50a7aa768489234ec56c3279fa1c4f7655d0a2a6 Thanks for the feedback